### PR TITLE
Add invoke task for listing various lib versions

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Add invoke task for listing various lib versions

### Motivation

- Revive @dineshg13's PR https://github.com/DataDog/datadog-agent/pull/20100 (slightly modified)
- https://datadoghq.atlassian.net/browse/ADXT-62

### CODE REVIEW

File: tasks/go.py [hunk 1]
```diff
@@ -650,3 +651,72 @@ def create_module(ctx, path: str, no_verify: bool = False):
...
+
+
+@task(iterable=['targets'])
+def mod_diffs(_, targets):
+    """
+    Lists differences in versions of libraries in the repo,
+    optionally compared to a list of target go.mod files.
+
+    Parameters:
+    - targets (str): Comma-separated list of paths to target go.mod files.
```
Comment (CelianR) [inline tasks/go.py:663]: If `targets` is iterable, then the list is not comma separated (I think it is better to not set it as iterable)

File: tasks/go.py [hunk 2]
```diff
@@ -650,3 +651,72 @@ def create_module(ctx, path: str, no_verify: bool = False):
...
+    # Find all go.mod files in the repo
+    all_go_mod_files = []
+    for module in DEFAULT_MODULES:
+        all_go_mod_files.append(os.path.join(module, 'go.mod'))
+
+    # Validate the provided targets
+    for target in targets:
+        if target not in all_go_mod_files:
+            print(f"Error: Target go.mod file '{target}' not found.")
+            return
```
Comment (CelianR) [inline tasks/go.py:674]: Could you print this on stderr / raise an Exception ?

File: tasks/go.py [hunk 3]
```diff
@@ -650,3 +651,72 @@ def create_module(ctx, path: str, no_verify: bool = False):
...
+            if relevant_paths:
+                print(f"Library {library} differs in:")
+                for version, paths in versions.items():
+                    intersecting_paths = set(paths).intersection(targets)
+                    if intersecting_paths:
+                        print(f"  - Version {version} in:")
+                        for path in intersecting_paths:
+                            print(f"    * {path}")
+        else:
+            if len(versions) > 1:
```
Comment (CelianR) [inline tasks/go.py:717]: **🥜 nitpick:** ‏Could you use `elif` in order to reduce a bit the indentation ?

**Review by CelianR — APPROVED** [review]

File: tasks/go.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/go.py:663]: I modified the code without changing the comment, modifying

File: tasks/go.py (hunk 2 above)
Comment (FlorentClarret) [reply tasks/go.py:674]: yup

File: tasks/go.py (hunk 3 above)
Comment (FlorentClarret) [reply tasks/go.py]: done

**Pull request merged**